### DISCUSSION TITLE
Thank you and a note about system load average

### DISCUSSION THREAD

**Discussion by tonymartino** [travisghansen/hass-pfsense#96] [Ideas]
Thanks for putting this together -- I'm a new user and am already getting some utility out of it. I was wondering why the system load average values were being displayed as a percentage in my entity card and found that you've set the _unit_of_measurement_ attribute to **%**. Please note that the load average is not a percentage value but rather a calculated value indicating the number of processes in the _running_ or _runnable_ states that are waiting for CPU time. The [Wikipedia article](https://en.wikipedia.org/wiki/Load_(computing)) has a very good explanation of this. Would you consider removing the unit_of_measure attribute from the load average entities?

Another important aspect of system load average is that it is not, in fact, CPU load average. Processes can be waiting for CPU time even when the system's CPUs are under no load at all. For example, something I've seen on many systems over the years (and a scenario the Wikipedia article discusses briefly) is NFS failures. A system can reach a load average in the hundreds when critical IO is blocked or slow. This leaves CPUs idling while processes in the running state can not be processed by the CPU because the system is waiting for date. Indeed, the [screenshot here](https://upload.wikimedia.org/wikipedia/commons/2/2d/Big-load.png) shows a system with ~10% CPU utilization and a load average reaching 108! So, I'm also hoping you'd consider changing the entity names from _%systemname% CPU Load Average*_  to _%systemname% System Load Average*_ or, more simply, _%systemname Load Average*_. Thanks for your consideration and thanks again for the integration!

**Comment by travisghansen** [comment]
Good notes! I’ll look into updating the code to better reflect reality.

**Reply by tonymartino** [reply]
Wonderful and much appreciated.

**Reply by travisghansen** [reply]
Is there any valid unit of measurement to be applied to these?

**Reply by tonymartino** [reply]
As far as I'm aware (and I just looked again), there is no unit of measure defined for load average. Further, no other tools that display load average specify a unit of measure. So I believe that the unit_of_measure attribute should be left empty.

**Reply by tonymartino** [reply]
Then again, the uptime man page says:
`System load averages is the average number of **processes** that are either in a runnable or uninterruptable state.  `
So _processes_ may be a valid unit of measure. My preference is none.

**Reply by travisghansen** [reply]
Sounds good.

**Reply by travisghansen** [reply]
Try `v0.5.8`: https://github.com/travisghansen/hass-pfsense/commit/3bae5882231043065114b6526ac755c59774bd7b

**Reply by tonymartino** [reply]
Haven't had a chance until. Today. Looks great, thank you!
